### PR TITLE
soteria: init at 0.1.0

### PR DESCRIPTION
**Pull request by NotAShelf**
- **sorteria: init at 0-unstable-2024-06-07**

## Description of changes

Adds a package for [Soteria](https://github.com/imvaskel/soteria), a Polkit authentication agent written in GTK.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See
  [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests)
    (look inside
    [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or
    [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in
    [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or
    [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are
    [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking)
    to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using
      `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all
      changes have to be committed, also see
      [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in
      `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md)
  (or backporting
  [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md)
  and
  [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md)
  Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or
        breaking
  - [ ] (Module updates) Added a release notes entry if the change is
        significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS
        module
- [ ] Fits
      [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/so/soteria/package.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  #  Takes advantage of nixpkgs manually editing PACKAGE_PREFIX by grabbing it from
+  #  the binary itself.
+  #  https://github.com/NixOS/nixpkgs/blob/9b5328b7f761a7bbdc0e332ac4cf076a3eedb89b/pkgs/development/libraries/polkit/default.nix#L142
+  #  https://github.com/polkit-org/polkit/blob/d89c3604e2a86f4904566896c89e1e6b037a6f50/src/polkitagent/polkitagentsession.c#L599
+  preBuild = ''
+    export POLKIT_AGENT_HELPER_PATH="$(strings ${polkit.out}/lib/libpolkit-agent-1.so | grep "polkit-agent-helper-1")"
+  '';
+
+  meta = {
+    description = "A Polkit authentication agent written in GTK designed to be used with any desktop environment.";
```
Comment (Sigmanificient) [inline pkgs/by-name/so/soteria/package.nix:50]:
```suggestion
    description = "Polkit authentication agent written in GTK designed to be used with any desktop environment";
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]
small nit about description guidelines

File: pkgs/by-name/so/soteria/package.nix (hunk 1 above)
Comment (NotAShelf) [reply pkgs/by-name/so/soteria/package.nix:50]: Ah right, the derivation was written before the indefinite article rule was added.

I'll fix it shortly.

**Review by itslychee — APPROVED** [review]

**Review by imvaskel — APPROVED** [review]

**Comment by NotAShelf** [comment]
Addressed review comments, and bumped the source to use the new v0.1.0 tag upstream released.

File: pkgs/by-name/so/soteria/package.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    pango
+  ];
+
+  # From upstream packaging:
+  #  Takes advantage of nixpkgs manually editing PACKAGE_PREFIX by grabbing it from
+  #  the binary itself.
+  #  https://github.com/NixOS/nixpkgs/blob/9b5328b7f761a7bbdc0e332ac4cf076a3eedb89b/pkgs/development/libraries/polkit/default.nix#L142
+  #  https://github.com/polkit-org/polkit/blob/d89c3604e2a86f4904566896c89e1e6b037a6f50/src/polkitagent/polkitagentsession.c#L599
+  preBuild = ''
+    export POLKIT_AGENT_HELPER_PATH="$(strings ${polkit.out}/lib/libpolkit-agent-1.so | grep "polkit-agent-helper-1")"
```
Comment (SuperSandro2000) [inline pkgs/by-name/so/soteria/package.nix:49]: Just hardcode /run/wrappers/bin/polkit-agent-helper-1

Comment (NotAShelf) [reply pkgs/by-name/so/soteria/package.nix:49]: Is it *guaranteed* to be there?

**Review by imvaskel — APPROVED** [review]

File: pkgs/by-name/so/soteria/package.nix (hunk 2 above)
Comment (JohnRTitor) [reply pkgs/by-name/so/soteria/package.nix:49]: If polkit is enabled, yes.

File: pkgs/by-name/so/soteria/package.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  meta = {
+    description = "Polkit authentication agent written in GTK designed to be used with any desktop environment";
+    homepage = "https://github.com/ImVaskel/soteria";
+    license = lib.licenses.asl20;
+    mainProgram = "soteria";
+    maintainers = with lib.maintainers; [
+      NotAShelf
+      lychee
+    ];
```
Comment (JohnRTitor) [inline pkgs/by-name/so/soteria/package.nix:60]:
```suggestion
    ];
    
    inherit (polkit.meta) platforms;
```

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 330682` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>soteria</li>
  </ul>
</details>

File: pkgs/by-name/so/soteria/package.nix (hunk 2 above)
Comment (NotAShelf) [reply pkgs/by-name/so/soteria/package.nix:49]: Likewise.

**Review by JohnRTitor — APPROVED** [review]
LGTM, ran successfully

**Comment by Aleksanaa** [comment]
@ofborg build soteria

**Pull request merged**